### ISSUE TITLE
[Reboot] Support for transparent button backgrounds in firefox

### ISSUE THREAD

**Issue by oleersoy** [twbs/bootstrap#18320]
Hi,

[SuitCSS's base module](https://github.com/suitcss/base/blob/master/lib/base.css) contains the following snippet to support focus styles in firefox when the background is transparent:
``` css

/**
 * Work around a Firefox/IE bug where the transparent `button` background
 * results in a loss of the default `button` focus styles.
 */

button:focus {
  outline: 1px dotted;
  outline: 5px auto -webkit-focus-ring-color;
}
```

Would this be appropriate for bootstrap to include?

Cheers,
Ole

**Comment by oleersoy** [comment]
SuitCSS Base also includes this:

``` css

/**
 * Suppress the focus outline on elements that cannot be accessed via keyboard.
 * This prevents an unwanted focus outline from appearing around elements that
 * might still respond to pointer events.
 */

[tabindex="-1"]:focus {
  outline: none !important;
}

```

**Comment by cvrebert** [comment]
@oleersoy Open a separate issue please.

**Comment by oleersoy** [comment]
Done - I deleted the original - here's the link to the new one:
https://github.com/twbs/bootstrap/issues/18330

**Comment by cvrebert** [comment]
Relevant git blame: https://github.com/suitcss/base/commit/88853f550c7cb421852f9ad0aa70e896e7e92f82

**Comment by mdo** [comment]
Is this still applicable to the versions of IE and FF we support? I see no details on what that note applies to after a quick look around SUIT.

**Comment by oleersoy** [comment]
Short answer yes.  Here's how I tested it using firefox version 42 via the SUIT base test case:

``` console
git clone https://github.com/suitcss/base
cd base
npm run setup
npm run build
npm run build-test
firefox test/index.html
```
Look at test `6.2 has focus styles (in Firefox and IE)`.  Open the firefox developer tools (`ctrl-shift-I`).  Click on the style tab.  Delete the following css (lines 568 - 579):

```css
button:focus {
  outline: 1px dotted;
  outline: 5px auto -webkit-focus-ring-color;
}
```

Now attempt to focus the test button.

**Comment by cvrebert** [comment]
Confirmed in Firefox 44.0a2 (2015-12-06).
Plunk: http://plnkr.co/edit/lCcQce46VJE1NWVMHJV6?p=preview

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://run.plnkr.co/plunks/lCcQce46VJE1NWVMHJV6/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W001](https://github.com/twbs/bootlint/wiki/W001): `<head>` is missing UTF-8 charset `<meta>` tag
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* line 138, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by mdo** [comment]
Fixed in c9eb483.
